docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -635,310 +960,179 @@ public class CustomSecurityIdentityAugmentor implements SecurityIdentityAugmento
+    }
+}
 ----
 
-Using `client_secret_jwt` or `private_key_jwt` authentication methods ensures that no client secret goes over the wire.
+=== Propagating tokens to downstream services
 
-==== Additional JWT authentication options
+For information about Authorization Code Flow access token propagation to the downstream services, see the xref:security-openid-connect-client-reference.adoc#token-propagation[Token Propagation] section.
 
-If `client_secret_jwt`, `private_key_jwt` authentication methods are used or an Apple `post_jwt` method is used, then the JWT signature algorithm, key identifier, audience, subject and issuer can be customized, for example:
 
-[source,properties]
-----
-# private_key_jwt client authentication
+== Integration considerations
 
-quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
-quarkus.oidc.client-id=quarkus-app
-quarkus.oidc.credentials.jwt.key-file=privateKey.pem
+Your OIDC backed application integrates in an environment such as calling from a single page application, using well known OIDC providers etc. 
+This section discusses these considerations.
 
-# This is a token key identifier 'kid' header - set it if your OpenID Connect provider requires it.
-# Note if the key is represented in a JSON Web Key (JWK) format with a `kid` property then
-# using 'quarkus.oidc.credentials.jwt.token-key-id' is not necessary.
-quarkus.oidc.credentials.jwt.token-key-id=mykey
+=== Single-page applications
 
-# Use RS512 signature algorithm instead of the default RS256
-quarkus.oidc.credentials.jwt.signature-algorithm=RS512
+Check if implementing SPAs the way it is suggested in the xref:security-oidc-bearer-token-authentication-concept.adoc#single-page-applications[Single-page Applications for Service Applications] section can meet your requirements.
 
-# The token endpoint URL is the default audience value, use the base address URL instead:
-quarkus.oidc.credentials.jwt.audience=${quarkus.oidc-client.auth-server-url}
+If you prefer to use SPA and JavaScript API such as `Fetch` or `XMLHttpRequest`(XHR) with Quarkus web applications, be aware that OpenID Connect Providers may not support CORS for Authorization endpoints where the users are authenticated after a redirect from Quarkus. This will lead to authentication failures if the Quarkus application and the OpenID Connect Provider are hosted on the different HTTP domains/ports.
 
-# custom subject instead of the client id :
-quarkus.oidc.credentials.jwt.subject=custom-subject
+In such cases, set the `quarkus.oidc.authentication.java-script-auto-redirect` property to `false` which will instruct Quarkus to return a `499` status code and `WWW-Authenticate` header with the `OIDC` value. The browser script also needs to be updated to set `X-Requested-With` header with the `JavaScript` value and reload the last requested page in case of `499`, for example:
```

Comment:
```suggestion
In such cases, set the `quarkus.oidc.authentication.java-script-auto-redirect` property to `false`, which will instruct Quarkus to return a `499` status code and a `WWW-Authenticate` header with the `OIDC` value. 
You must also update the browser script to set the `X-Requested-With` header with the `JavaScript` value and reload the last requested page in case of a `499` status code.
For example:
```